src/main/java/com/softserveinc/dokazovi/service/impl/UserServiceImpl.java

```diff
@@ -97,7 +97,11 @@ public Page<UserEntity> findAll(Pageable pageable) {
      */
     @Override
     public UserDTO findExpertById(Integer userId) {
-        return userMapper.toUserDTO(userRepository.findById(userId).orElse(null));
+        AuthorEntity author = authorRepository.findById(userId).orElse(null);
```

Comment:
```suggestion
        AuthorEntity author = authorRepository.findById(userId).orElseThrow(() -> new EntityNotFoundException("Author not found"));;
```
And you can remove if-statement. 
This is a more concise way to achieve the same outcome.